zap/src/main/java/org/parosproxy/paros/core/scanner/HostProcess.java

```diff
@@ -363,6 +367,45 @@ public void run() {
         }
     }
 
+    private boolean bestNodeForUrl(StructuralNode node) {
+
+        if (!hasSamePropertiesAsRequestCreatedToFillSiteMap(node)) {
+            return true;
+        }
+
+        StructuralNode parentNode;
+        try {
+            parentNode = node.getParent();
+        } catch (DatabaseException e) {
+            return true;
+        }
+
+        if (parentNode == null) {
+            return true;
+        }
+
+        String nodeURI = node.getURI().getEscapedURI();
+        String nodeMethod = node.getMethod();
+        Iterator<StructuralNode> siblingsIt = parentNode.getChildIterator();
+        while (siblingsIt.hasNext()) {
+            StructuralNode sibling = siblingsIt.next();
+            String childURI = sibling.getURI().getEscapedURI();
+            if (!sibling.isSameAs(node)
+                    && sibling.getMethod().equals(nodeMethod)
+                    && childURI.equals(nodeURI)
+                    && sibling.isLeaf()
+                    && !hasSamePropertiesAsRequestCreatedToFillSiteMap(sibling)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean hasSamePropertiesAsRequestCreatedToFillSiteMap(StructuralNode node) {
+        return node.getMethod().equals(HttpRequestHeader.GET)
+                && node.getHistoryReference().getHistoryType() == TYPE_TEMPORARY;
```

Comment:
Instead of importing and using you can simply reference it statically.
```suggestion
                && node.getHistoryReference().getHistoryType() == HistoryReference.TYPE_TEMPORARY;
```

Reply:
thx, I  know but I was using it more than once and lines were getting to big to fit in one line.